test/e2e/api-specs/MultichainAuthorizationConfirmationErrors.ts

```diff
@@ -0,0 +1,161 @@
+      addToQueue({
+        name: 'afterRequest',
+        resolve,
+        reject,
+        task: async () => {
+          if (this.errorCodesToHitCancel.includes(call.expectedResult?.code)) {
+            try {
+              await this.driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+              const text = 'Cancel';
+
+              await this.driver.findClickableElements({
+                text: 'Cancel',
+                tag: 'button',
+              });
+
+              const screenshot = await this.driver.driver.takeScreenshot();
+              call.attachments = call.attachments || [];
+              call.attachments.push({
+                type: 'image',
+                data: `data:image/png;base64,${screenshot}`,
+              });
+              await this.driver.clickElement({ text, tag: 'button' });
+              // make sure to switch back to the dapp or else the next test will fail on the wrong window
+              await switchToOrOpenDapp(this.driver);
+            } catch (e) {
+              console.log(e);
+            }
+          }
+        },
+      });
+    });
+  }
+
+  // get all the confirmation calls to make and expect to pass
+  getCalls(__: unknown, method: MethodObject) {
+    const calls: Call[] = [];
+    const isMethodAllowed = this.only ? this.only.includes(method.name) : true;
+    if (isMethodAllowed) {
+      if (method.errors) {
+        method.errors.forEach((err) => {
+          const unsupportedErrorCodes = [5000, 5300, 5301];
+          const error = err as ErrorObject;
+          if (unsupportedErrorCodes.includes(error.code)) {
+            return;
+          }
+          let params: Record<string, unknown> = {};
+          switch (error.code) {
+            case 5100:
+              params = {
+                requiredScopes: {
+                  'eip155:10124': {
+                    methods: ['eth_signTypedData_v4'],
+                    notifications: [],
+                  },
+                },
+              };
+              break;
+            case 5101:
+              params = {
+                requiredScopes: {
+                  'eip155:1': {
+                    methods: ['foo'],
+                    notifications: [],
+                  },
+                },
+              };
+              break;
+            case 5102:
+              params = {
+                requiredScopes: {
+                  'eip155:1': {
+                    methods: [],
+                    notifications: ['potato'],
+                  },
+                },
+              };
+              break;
+            case 5302:
+              params = {
+                requiredScopes: {
+                  'eip155:1': {
+                    methods: ['eth_signTypedData_v4'],
+                    notifications: [],
+                  },
+                },
+                sessionProperties: {},
+              };
+              break;
+            default:
+              break;
```

Comment:
do we need to do anything special if we don't match an error in this switch statement?